### ISSUE TITLE
Using --gmusic-playlist exits with error 'maximum recursion depth exceeded in cmp'

### ISSUE THREAD

**Issue by jotjotjes** [tizonia/tizonia-openmax-il#519]
### Description

I try to play a google music playlist.
Tizonia stops for a long time after connecting and retrieving the playlist and exits with the error: `(RuntimeError) : maximum recursion depth exceeded in cmp`

### Steps to Reproduce

1.  tizonia --gmusic-user <name>@gmail.com --gmusic-password <password> --gmusic-device-id <device-id> --gmusic-playlist 'Psycho'
2. wait for some time

**Expected behavior:** The first track of the playlist should be played

**Actual behaviour:** Tizonia exits with an error message

**Reproduces how often:** 100% (also tested with different playlists)

#### Tizonia

tizonia 0.15.0. Copyright (C) 2018 Juan A. Rubio
Debug Info:
	    * [Boost 1_58]
	    * [TagLib 1.9.1]
	    * [MediaInfoLib - v0.7.82]

#### OS

NAME="Ubuntu"
VERSION="18.04.1 LTS (Bionic Beaver)"

### Installation Method Used

Snap package

#### Full Output of Session

```
# > tizonia --gmusic-user <name>@gmail.com --gmusic-password <password> --gmusic-device-id <device-id> --gmusic-playlist 'Psycho'
tizonia 0.15.0. Copyright (C) 2018 Juan A. Rubio
This software is part of the Tizonia project <http://tizonia.org>

[Google Play Music] [Connecting] : '<name>@gmail.com'.
[Google Play Music] [Authenticating] : 'with cached auth token'
[Google Play Music] [Retrieving library] : '<name>@gmail.com'. 
[Google Play Music] [Playlist] 'Thumbs Up' (125 tracks).
[Google Play Music] [Playlist] 'Beethoven 2. Symphonie' (4 tracks).
[Google Play Music] [Playlist] 'Beethoven 5. Symphonie' (4 tracks).
[Google Play Music] [Playlist] 'Beethoven 9. Symphonie' (5 tracks).
[Google Play Music] [Playlist] 'Psycho' (3 tracks).
[Google Play Music] [Playlist] 'Gorecki 3. Symphonie' (3 tracks).
[Google Play Music] [Playlist] 'stg2014' (92 tracks).
[Google Play Music] [Playlist] 'indiepitch' (327 tracks).
[Google Play Music] [Playlist] 'Die ruhige Rille' (37 tracks).
[Google Play Music] [Playlist] 'neuebesen' (505 tracks).
[Google Play Music] [Playlist] 'Zwischenlager' (126 tracks).
[Google Play Music] [Playlist] 'Mag ich alles' (240 tracks).
[Google Play Music] Playing 'Psycho'.
[Google Play Music] [Track] 'Psychiater' by 'Funny Van Dannen'.
[Google Play Music] [Track] 'Finger weg von meiner Paranoia' by 'Element of Crime'.
[Google Play Music] [Track] 'American Beauty / American Psycho' by 'Fall Out Boy'.
[Google Play Music] [Tracks in queue] '3'.
[Google Play Music] (RuntimeError) : maximum recursion depth exceeded in cmp

tizonia exiting (OMX_ErrorInsufficientResources).

 [OMX.Aratelia.audio_source.http:port:0]
 [OMX_ErrorInsufficientResources]

# >
```

**Comment by tizonia** [comment]
Hi!, thanks for reporting.

The first thing to ask is, do you have up-to-date Python dependencies?. See Step 2 in the Upgrade section:
https://github.com/tizonia/tizonia-openmax-il#upgrade-debian--ubuntu--raspbian

If after updating, you still face this issue, please export this environment variable before running the command:
```
export TIZONIA_GMUSICPROXY_DEBUG=1
```
and then it would be useful if you uploaded here the logs produced.

Thanks!

**Comment by jotjotjes** [comment]
The problem still persist after the updates.

This is the very long log file:
[gmusicproxy.log](https://github.com/tizonia/tizonia-openmax-il/files/2425282/gmusicproxy.log)

**Comment by tizonia** [comment]
Thanks for uploading the log.

I think there are two issues here. One looks clear. When none of the items in the playlist is actually playable, the gmusic proxy enters into an infinite recursion, which is definitely something that needs fixing.

The second issue I don't understand yet. It looks like none of the songs on those playlists is actually playable. Google is returning a 403 during URL retrieval. Any ideas on why this might be?. Are/were you a paid subscriber?

**Comment by jotjotjes** [comment]
No, I'm not a subscriber. I use google music only to listen to my uploaded music files. Does this mean I can listen to my music only within the web interface?

**Comment by tizonia** [comment]
Sorry for the delay in my replies. You should be able to play your music with a non-paid account. At least, this used to be the case. I need to figure out if something has changed at the API level. Tizonia uses [gmusicapi](https://github.com/simon-weber/gmusicapi), which is an unofficial API (i.e. reversed engineered). So sometimes things break when Google decides to change a bit the API. And those changes many times get reworked in the next release of gmusicapi.

So I'll investigate.
